Split ui from backend2

jrafanie opened the pull request:
WIP

Commit:
archived entitites of an ems should be destroyed with the ems

jrafanie committed:
quiesce_time_allowance per worker is not needed

We already check quiesce_loop_timeout and kill all workers if the loop
timeout is reached.  It doesn't matter what quiesce_time_allowance is
for a worker if the quiesce_loop_timeout is lower, since the loop
timeout will expire first and cause all workers to be killed.

It doesn't seem necessary to have two timeouts and doing it at a worker
level seems too complex.

The original commit and bugzids don't indicate this per worker class
timeout as a feature so it must have been added as something we might
use in the future.

Below is the original commit found by NickL :tada:

commit aa2f40fa4504bae0854e6ac194b89a43845e9934
Author: Joe Rafaniello <jrafanie@redhat.com>
Date:   Tue Mar 29 23:21:47 2011 +0000

    Add a synchronous stop of the server and call it from EvmApplication(used by rake evm:stop/restart and black console)
    Add quiesce logic to allow queue worker types to exit gracefully, other types are killed, and only continue shutdown of the server when they're all stopped or killed.
    If provided, use :quiesce_loop_timeout in worker_monitor section of configuration
    If provided, use :quiesce_time_allowance in each worker's settings
    bugzid:11862,11825,11760,11719

gberginc committed:
Add IOPS and encrypted columns to CloudVolume

Amazon EBS provides information about volume's IOPS and encryption
status as part of the cloud volume model so this patch extends the
existing cloud model to include these two attributes.

Commit:
update MiqExpression::Field to return col_name

Commit:
add parse_report_field specs

Commit:
add instance method to field and tag

djberg96 committed:
Move resource_groups relationship to VmOrTemplate model.

Modified vms method, added templates method.

Added resource_group specs.

Updated ResourceGroup specs to reflect model changes.

Rely on default scope for vms method.

Remove tests that were only testing Rails.

Update templates association and specs.

syncrou committed:
Add require 'rest_client'

The rebuild_provision_request tool can fail to run if rest-client is not required
This change guarantees the gem is loaded before running the command
Also adds the dependency rest-client back into the Gemfile to let the
tool run on its own

NickLaMuro committed:
Adds lib/manageiq.rb

Adds two methods to a global ManageIQ module:

  ManageIQ.env:   Functions the same as Rails.env, but falls back to an
                  equivalent implementation when Rails isn't present
  ManageIQ.root:  Functions the same as Rails.root, but falls back to an
                  equivalent implementation when Rails isn't present

Tests are done in such a way that if Rails changes it's implementation
that causes a difference in what we are seeing here, hopefully these
tests will catch it.

Commit:
extract google provider

Commit:
Added google provider to gemfile

Commit:
Removed Google GCE specific gems

Commit:
Removed Microsoft SCVMM provider to a separate gem.

Commit:
Removed Foreman provider to a separate gem.

jrafanie committed:
Move dependencies to bundler groups that workers require

Use an environment variable for now to default which groups are loaded.
In workers, override this variable with what individual workers need.

Delay requires or move them to where they are needed.
Please enter the commit message for your changes. Lines starting

hayesr committed:
Compile and write stylesheets for PDF rendering

https://bugzilla.redhat.com/show_bug.cgi?id=1447940

AndreyMenezes committed:
Add physical_server_id to event streams table

yrudman committed:
do not delete report when linked MiqTask deleted

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1450272
Changes:
- removed :dependent => :destroy on miq_report_result association in MiqTask
- refactor MiqReportResult#status to return 'Complete' if task asociated with report deleted
- added test coverage for MiqReportResult#status
- remove not used virtual_delegate to MiqTask#state_or_status

mfeifer committed:
Changelog 61

101 PRs
updated heading styles

[skip ci]

nimrodshn committed:
move openshift related settings

nimrodshn committed:
move image inspector settings

chrisarcand committed:
Move setup objects to lets

chrisarcand committed:
Add logging note

After my failed attempt to not do the hacking we do with the logger.

This isn't easily fixed because there are assertions about provider code
still coupled to this repo, and it's not worth losing test coverage
over right now.

chrisarcand committed:
More moving to helpers

chrisarcand committed:
Remove useless stats_counter vars

chrisarcand committed:
Wrap historical tests in a context

chrisarcand committed:
Move realtime capture into it's own nicely wrapped context

chrisarcand committed:
Cleanup

chrisarcand committed:
Missed one

chrisarcand committed:
Put expected timestamp range all in one spot

chrisarcand committed:
Down Rubocop, down boy.

imtayadeway committed:
Only route PUT/PATCH/DELETE to collection members

imtayadeway committed:
Remove unreachable raise

imtayadeway committed:
id arg is not optional for delete_subcollection_resource

imtayadeway committed:
Delete unreachable raise

NickLaMuro commented on Gemfile:
```diff
@@ -95,11 +67,94 @@ gem "ruport",                         "=1.7.0",                       :git => "h
+  manageiq_plugin "manageiq-providers-hawkular"
+end
+
+group :kubernetes do
```
Since `manageiq-providers-openshift` pretty much inherits from `manageiq-providers-kubernetes`, you might just want to make this `group :kubernetes, :openshift do`.

NickLaMuro reviewed (commented):
I think this is my only concern at the moment, and it is pretty small.

nimrodshn committed:
removed missing settings

.

refactoring

move back container settings

gildub committed:
Add cloud subnet REST API

aufi committed:
Remove Openstack events Settings file

File config/settings.yml contain list of events including Openstack specific Events.

Moving such events into manageiq-providers-openstack repo.

isimluk committed:
Do not calculate useless group metrics

https://bugzilla.redhat.com/show_bug.cgi?id=1455690
https://bugzilla.redhat.com/show_bug.cgi?id=1454456

 - cpu is useless as it counts usagemhz and vm_numvcpus together
 - cpu_cores is useless as it only reports cpu_cores
 - memory is useless as counts used and available together
 - net_io is useless as it only reports net_ui_rate_avg
 - disk_io is useless as it only reports disk_usage_rate_avg
 - storage is useless as it counts used and allocated together
 - fixed is useless as it counts all the fixed metrics together
   (that equals to 4 for each day, not useful)

isimluk committed:
Replace metric_keys method with metric_key method

It used to return array with single item, now it just returns that
single item.

Less object creation -> less fat in the code -> performance goes up.

isimluk committed:
Do not offer report columns that are useless

See https://github.com/ManageIQ/manageiq/pull/15260 to learn why this is
not useful.

https://bugzilla.redhat.com/show_bug.cgi?id=1455690
https://bugzilla.redhat.com/show_bug.cgi?id=1454456

imtayadeway committed:
Clarify else condition in routes.rb

The only other option here is `:post`, so make that explicit.

imtayadeway committed:
Prefer create_list to x.times { create }

imtayadeway committed:
Put a space around hashrocket

imtayadeway committed:
Inline before block

imtayadeway committed:
Remove unnecessary test fixtures

imtayadeway committed:
Move the .purge_one_batch test to its own example group

imtayadeway committed:
Don't create 2 sessions where 1 is needed

imtayadeway committed:
Collapse multiline method call into one line

imtayadeway committed:
Use change(&block) to verify before/after

imtayadeway committed:
Mock the user double to assert that it receives the logoff message

imtayadeway committed:
Remove some test duplication

lpichler committed:
Allow matching via descendants even it is the only filter

lpichler committed:
Allow matching via descendants for CloudNetwork via ExtManagementSystem(=in that case NetworkManager)

hayesr committed:
Format trend max cpu usage rate with percent
https://bugzilla.redhat.com/show_bug.cgi?id=1451132

gtanzillo committed:
Ensure report columns serialized as hashes have symbolized keys before importing

This fixes a bug where a report imported via the Rest API, as json, fails to generate because the keys
in columns serialized as hashes are received as strings.

https://bugzilla.redhat.com/show_bug.cgi?id=1456742

AparnaKarve committed:
Add entries for Physical Server

hayesr committed:
Clean up pdf_generator_spec

chessbyte committed:
Merge pull request #15071 from juliancheal/extract_foreman

Removed Foreman provider to a separate gem.

chessbyte committed:
Merge pull request #15103 from juliancheal/extract_scvmm

Extract scvmm

gtanzillo committed:
Merge pull request #15134 from yrudman/fix-deleting-report-when-task-deleted

Do not delete report if task associated with this report deleted

chessbyte committed:
Merge pull request #15043 from juliancheal/extract_google

Removed Google provider to a separate gem.

Fryguy committed:
Extract ArRegion into activerecord-id_regions gem

ArRegion still exists, but only for the ManageIQ specific changes.  This
commit also adds specs for those specific features in ArRegion to ensure
they weren't broken in the extraction.

chessbyte committed:
Merge pull request #15272 from hayesr/trend_max_pct

Format trend max cpu usage rate with percent

chessbyte committed:
Merge pull request #15131 from hayesr/compiled_pdf_css

Compile and write stylesheets for PDF rendering

chessbyte committed:
Merge pull request #15273 from gtanzillo/report-import-symbolize-keys

Ensure report columns serialized as hashes have symbolized keys before importing

chessbyte committed:
Merge pull request #15268 from imtayadeway/refactor/session-specs

Refactor Session specs

chessbyte committed:
Merge pull request #15275 from AparnaKarve/add_phy_server_en_yml

Add entries for Physical Server

chessbyte committed:
Merge pull request #15219 from chrisarcand/refactor-ci-mixin-metric-spec

Refactor metrics/ci_mixin/capture spec

chessbyte committed:
Merge pull request #15178 from nimrodshn/move_openshift_related_settings

Moving openshift related settings

chessbyte committed:
Merge pull request #14359 from zeari/delete_archived_entities_on_ems_delete

Delete archived entities when a container manager is deleted

mzazrivec committed:
Add pretty model name for physical server

bdunne committed:
Merge pull request #14948 from djberg96/resource_group

Move ResourceGroup relationship into VmOrTemplate model

gtanzillo committed:
Merge pull request #15271 from lpichler/allow_matching_descendants_for_cloud_networks

Allow matching via descendants for CloudNetworks (via network manager)

syncrou committed:
Add Enhanced Debug level support

https://www.pivotaltracker.com/story/show/146008959

jrafanie committed:
Merge pull request #15281 from Fryguy/ar_region_extraction

Extract ArRegion into activerecord-id_regions gem

gmcculloug committed:
Merge pull request #15288 from syncrou/add_verbosity_to_ansible_playbook

Add Enhanced Debug level support

abellotti committed:
Merge pull request #15248 from gildub/cloud_subnet_API_initial

Add cloud subnet REST API

abellotti committed:
Merge pull request #15234 from imtayadeway/api/refactor/put-patch-delete-routes

Only route PUT/PATCH/DELETE for collection member routes

Fryguy committed:
Merge pull request #15133 from AndreyMenezes/XCS-120

Add lenovo attributes to event streams table

Fryguy committed:
Merge pull request #14704 from gberginc/add_iops_and_encrypted_to_cloud_volume

Add IOPS and encrypted columns to CloudVolume

chessbyte committed:
Merge pull request #15283 from mzazrivec/add_model_name_for_physical_server

Add pretty model name for physical server

Fryguy committed:
Remove dependency of UuidMixin in migrations

chessbyte committed:
Merge pull request #15294 from Fryguy/remove_uuidmixin_from_migrations

Remove dependency of UuidMixin in migrations

cben committed:
Add model associations for containers graph refresh

agrare committed:
Merge pull request #15298 from cben/grafresh-associations

Add model associations for containers graph refresh

bdunne committed:
Merge pull request #14985 from syncrou/add_require_for_rest_client

Add require 'rest-client'

chessbyte committed:
Merge pull request #15258 from aufi/remove_openstack_events

Remove Openstack events from Settings file

chessbyte committed:
Add dependency on rubyzip gem

chessbyte committed:
Add dependency on openscap gem

chessbyte committed:
Merge pull request #15179 from nimrodshn/move_kubernetes_related_settings

Moving kubernetes related settings

imtayadeway committed:
Put time back the way we found it

These tests use `Timecop.travel` without `Timecop.return`, causing
another test in another part of the suite to fail. To reproduce:

```
SPEC_OPTS=--seed=30045 bundle exec rspec ./spec/models/job_spec.rb[1:1:7:1:1] ./spec/requests/api/logging_spec.rb[1:1:4]
```

Fixes https://github.com/ManageIQ/manageiq/issues/15290

jrafanie committed:
Merge pull request #15020 from NickLaMuro/miq_helper

Adds MiqHelper

jrafanie committed:
Fix some minor typos from #15020

[skip ci]

bdunne committed:
Merge pull request #15300 from chessbyte/add_openscap_dependency

Add dependency on openscap gem

bdunne committed:
Merge pull request #15299 from chessbyte/add_rubyzip_dependency

Add dependency on rubyzip gem

bdunne committed:
Merge pull request #15302 from jrafanie/fix_typos

Fix some minor typos from #15020

carbonin committed:
Handle additional case for /etc/tower/SECRET_KEY

Previously if we had a value in the database, but the file
didn't exist on the filesystem .configured? would raise an error
when it should really just return false and we will write out
the value from the database to the filesystem.

https://bugzilla.redhat.com/show_bug.cgi?id=1439783
https://bugzilla.redhat.com/show_bug.cgi?id=1458886

gtanzillo committed:
Merge pull request #14905 from jntullo/bz/report_styles

Correct field names for reports

gtanzillo committed:
Merge pull request #14643 from jrafanie/delete_unneeded_quiesce_time_allowance_per_worker

quiesce_time_allowance per worker is not needed

gtanzillo committed:
Merge pull request #15260 from isimluk/do-not-report-useles-group-metric

Do not calculate useless group metrics

gtanzillo committed:
Merge pull request #15261 from isimluk/do-not-offer-useless-metrics

Do not offer report columns that are useless

AlonaKaplan committed:
ovirt-networking: using profiles

This patch fixes some issues in oVirt vm provision network tab.
1. Changing the 'vLan' label to 'Network'
2. Displaying the vnic profiles instead of the host/hosts networks (only
profiles configured in the cluster).
3. Using OvirtSDK4.

Changes relevant to version 4 only.

isimluk committed:
Merge pull request #15301 from imtayadeway/bug/fix-interacting-time-tests

Put time back the way we found it

agrare committed:
Merge pull request #14991 from AlonaKaplan/ovirt_networking_2

ovirt-networking: using profiles

carbonin committed:
Remove the secret key from the database when the setup fails

This will force `.configured?` to false the next time `.start` is run
allowing us to retry the configuration.

Before this change, users would have to blank the SECRET_KEY file
on the filesystem to force a retry.

https://bugzilla.redhat.com/show_bug.cgi?id=1439783
https://bugzilla.redhat.com/show_bug.cgi?id=1458886

Commit:
Removed SCVMM Data as moved to manageiq-providers-scvmm

Fryguy committed:
Merge pull request #15168 from mfeifer/masterchangelog61

Changelog 61

tinaafitz committed:
Log zone(q_options) when raising retirement event.

https://bugzilla.redhat.com/show_bug.cgi?id=1459236

jrafanie committed:
Merge pull request #15313 from carbonin/handle_setup_playbook_failure_better

Handle setup playbook failure better

gmcculloug committed:
Merge pull request #15317 from tinaafitz/retirement_zone_log_message

Log zone(q_options) when raising retirement event.

Commit:
Fixed alignment of VCR method

We keep fixing this in the providers so this will fix at source.

blomquisg committed:
Merge pull request #15314 from juliancheal/remove_scvmm_data

Removed SCVMM Data as moved to manageiq-providers-scvmm

bdunne committed:
Merge pull request #15319 from juliancheal/fixed_vcr_alignment

Fixed alignment of VCR method

jrafanie committed:
spawn instead of fork

jrafanie committed:
Debugging of process output and loaded features.
